### PR TITLE
eliminate infinite loop in case of a spurious file path

### PR DESCRIPTION
**Pull request by szemere**
Instead of a simple boolean result, file_opener_open_fd will return
that the error was temporary or permanent. (Currently spurious paths
is the only permanent error.)
If the error is permanent we can do nothing about it, so we skip the
logwriter_reopen method too, thus there will be no further attempt to
open the file, and the log message will be skipped by the file
destination.

WIP:
- [x] perftest
- [x] news file

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/affile/affile-dest.c [hunk 1]
```diff
@@ -168,6 +169,9 @@ affile_dw_reopen(AFFileDestWriter *self)
     }
   else
     {
+      if (open_result == FILEOPENER_RESULT_ERROR_PERMANENT)
```
Comment (bazsi) [inline modules/affile/affile-dest.c:172]: this should be an "else if (open_result == FILEOPENER_RESULT_ERROR_PERMANENT) {} else {}" block, e.g.
the else and if should be combined, makes the flow more readable.

File: modules/affile/file-opener.h [hunk 2]
```diff
@@ -47,6 +47,12 @@ typedef struct _FileOpenerOptions
   gint create_dirs;
 } FileOpenerOptions;
 
+typedef enum {
+  FILEOPENER_RESULT_SUCCESS,
```
Comment (bazsi) [inline modules/affile/file-opener.h:51]: what do you think about an underscore between FILE and OPENER? those are separate terms, the filenames and other identifiers follow this convetion.

File: modules/affile/affile-dest.c (hunk 1 above)
Comment (szemere) [reply modules/affile/affile-dest.c]: Thank You!

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
@kira-syslogng do perftest

**Comment by kira-syslogng** [comment]
Kira-performance-test: Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
I've used the following config to test the fix:

```
@version: 3.26

options {
  time-reopen(1);
};

log {
  source { example-msg-generator(num(1) template("../../path")); };
  destination { file("/tmp/$MSG.txt"); };
};
```

File: news/bugfix-3230.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+`affile`: eliminate infinite loop in case of a spurious file path
+
+If the template evaluation of a log message will result to a spurious
+path in the file destination, syslog-ng refuses to create that file.
+However the problematic log message was left in the msg queue, so
+syslogng was trying to create that file again in time-reopen periods.
```
Comment (MrAnno) [inline news/bugfix-3230.md:6]: syslog-ng

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**